modules/tinymce/src/plugins/image/test/ts/browser/ContextMenuTest.ts

```diff
@@ -36,6 +37,25 @@ describe('browser.tinymce.plugins.image.ContextMenuTest', () => {
     TinyAssertions.assertSelection(editor, [], 0, [], 1);
   });
 
+  it('TINY-9491: Opening context menus on a selected figure', async () => {
+    const editor = hook.editor();
+    editor.setContent(
+      '<div class="mce-toc" data-mce-toc="true" contenteditable="false">' +
+        '<h2 contenteditable="true">Table of Contents</h2>' +
+        '<ul>' +
+          '<li>' +
+            '<a href="#mcetoc_1gmi3gu5b11" data-mce-href="#mcetoc_1gmi3gu5b11">point </a>' +
+          '</li>' +
+        '</ul>' +
+      '</div>' +
+      '<h2 id="mcetoc_1gmi3gu5b12">point</h2>',
+      { format: 'raw' }
+    );
+    Mouse.contextMenuOn(TinyDom.body(editor), 'a');
+    await Waiter.pWait(250); // Give plenty of time for the context menu to appear
```

Comment:
Please use a proper waiter here that waits for a state to appear this just screams flaky tests to me.

Reply:
There is a pTriggerContext menu that can likely be used in this case.

Reply:
The test here is to make sure that nothing happens. Since nothing changes, we can't wait for a change. So I set it up to wait for the context menu to have a chance to appear, and then make sure it's not there. Not sure how to improve that otherwise.

Reply:
The menu item should be disabled or maybe even not showing at all since it makes little sense to have a button that does nothing. So then one could check for that have a waiter that waits for the context menu then verifies that the menu item is disabled or not there.

Reply:
Yes. The context menu does not appear. I can't wait for it to appear, because it won't, and I can't wait for it not to appear, because that will immediately pass the test, even if it was about to show up. So I set a period to wait for it to appear, and then check that it doesn't. I can't really see a better way of checking that nothing happens.

Reply:
Maybe for this test one can configure two context menu items, one should be presented there always (for ex. separator) and the Link item sometimes. Then wait for the context menu to always appeared and check for the availability of the Link item in it.

Reply:
Did as such.